code/modules/projectiles/gun_attachables.dm

```diff
@@ -121,7 +121,7 @@ Defined in conflicts.dm of the #defines folder.
 	*/
 	if(G.attachments[slot])
 		var/obj/item/attachable/A = G.attachments[slot]
-		A.Detach(G)
+		A.Detach(null, G)
```

Comment:
G should be a named arg without the null... we hate null args carlarc

Reply:
****just like i hate you